packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -222,13 +222,42 @@ abstract class SerializationManager {
     Object? object, {
     bool formatted = false,
   }) {
-    if (object is ProtocolSerialization) {
+    // detect List<? extends ProtocolSerialization>
+    if (object is List) {
+      object = _convertListForProtocol(object);
+    } else if (object is Map) {
+      // detect every ProtocolSerialization in map.values
+      object = _convertMapForProtocol(object);
+    } else if (object is ProtocolSerialization) {
       return encode(object.toJsonForProtocol(), formatted: formatted);
     }
-
     return encode(object, formatted: formatted);
   }
 
+  static List _convertListForProtocol(List list) {
```

Comment:
Question: Does it make more sense to add these as extensions on list instead? Then we could just call toJsonForProtocol on the list as well?